### DISCUSSION TITLE
.dir-locals.el for project specific consult commands

### DISCUSSION THREAD

**Discussion by nicolas-graves** [minad/consult#844] [Q&A] [answered]
I haven't found this in the README nor in the issues, if that's something you want to document, I use something like this for consult-find : 
```
((nil
  . ((eval . (require 'consult))
     (eval . (setq consult-find-args
                 (if (string-equal (eval (car (get 'consult-find-args 'standard-value))) consult-find-args)
                     (concat consult-find-args " -a -not ( -wholename */outputs/* -prune )")
                   consult-find-args))))))
```
For ripgrep and fd, we have .rgignore and .fdignore, but I couldn't find something similar for find. 

You can close the issue if you do not want to document this.

**Comment by minad** [comment]
Hi Nicolas, thanks for the suggestion! I have a bit of a hard time to decode what you are doing exactly here and what you want to achieve. I'd appreciate if you add a few more explanations. Do you think this should be documented in the README or the wiki? Peripheral/special configurations can also be documented in the wiki. Thanks again!

**Reply by nicolas-graves** [reply]
So basically, in a work project I have a folder which fills with junk I have to keep for some time, but I don't want to parse it with ripgrep or find the files with `consult-find` in it. For `fd` and `ripgrep`, we have .ignore files, but not for find (which I use, I don't really use consult-fd). 

What the .dir-locals.el does is to concat `" -a -not ( -wholename */outputs/* -prune )"` to `consult-find-args` (when used as a string). A simpler version added the suffix each time I opened a file, so I added a test to see if the default value was used. This is indeed in a `.dir-locals.el`.

**Comment by minad** [comment, marked as answer]
Rereading the issue again, I wonder why you are using `eval` instead of setting the `consult-find-args` directly via the `.dir-locals.el`. I think this is what confused me when I took a look at this, since it may be more complicated than necessary. You probably do this such that you can adjust the `consult-find-args` and don't have to override them completely. 

The idea of `consult-find-args` and the other similar variables is that you can build up the command dynamically. Your use case may be better configured as follows:

~~~ elisp
(defvar my-consult-find-auxillary-args nil)
(setq consult-find-args 
  '("find . -not ( -wholename */.* -prune )" my-consult-find-auxillary-args))
~~~

Then you can override `my-consult-find-auxillary-args` via .dir-locals.el. Note that I haven't tested this myself, but this approach should work in principle.

**Reply by nicolas-graves** [reply]
This could indeed work better. I'll test this in the coming days and confirm it matches my use case.
